### DISCUSSION TITLE
Migrating an UNO App from 4.8 to 5.0

### DISCUSSION THREAD

**Discussion by Seekuhnetz** [unoplatform/uno#15006] [Q&A]
Hello,

i'm very new to UNO and currently have to migrate an existing UNO App from 4.8 to 5.0.
The project was developed by someone else and i can not ask them.

They put the `NavigationView` in the `Shell.xaml` and their Shell is also a Page instead of an UserControl like in the new Template. They do not have a MainPage.
In any tutorial or sample project i can find the `NavigationView` is used in the `MainPage.xaml`. For example even in the [responsive shell tutorial](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Navigation/Advanced/HowTo-ResponsiveShell.html).
Is it the best practice to have a MainPage and use it for Navigation?

**Comment by jeromelaban** [comment]
You're likely looking for [`NavigationView` documentation](https://learn.microsoft.com/en-us/windows/apps/design/controls/navigationview#code-example), which explains the basics of this control. The responsive shell documentation is using Uno.Extensions navigation, which is likely not what you have in your original app.

Note that to migrate to uno 5, you'll need to follow [this migration documentation](https://platform.uno/docs/articles/migrating-to-uno-5.html) as well.

**Reply by Seekuhnetz** [reply]
I added this simple NavigationView to my MainPage:

```
<NavigationView>
    <NavigationView.MenuItems>
        <NavigationViewItem Content="Nav Item A"/>
        <NavigationViewItem Content="Nav Item B"/>
        <NavigationViewItem Content="Nav Item C"/>
    </NavigationView.MenuItems>

    <Frame x:Name="ContentFrame"/>
</NavigationView>
```
But the NavigationViewItems are cut off when the View is collapsed. Any idea why this is the case? I am testing on the stock uno template.
![grafik](https://github.com/unoplatform/uno/assets/155576794/af9c720e-b701-4459-a40a-a73d5c33690e)

**Reply by kazo0** [reply]
@Seekuhnetz could you provide a .zip of a sample app that is having this problem so I can test it out?

Is Uno.Material being used in this project?

**Reply by Seekuhnetz** [reply]
@kazo0 
> @Seekuhnetz could you provide a .zip of a sample app that is having this problem so I can test it out?

Yes, i just created one: [NavigationViewIssue.zip](https://github.com/unoplatform/uno/files/13890558/NavigationViewIssue.zip)

> Is Uno.Material being used in this project?

Yes, in the Template Wizard i checked "Material", "Theme Service" and "Import DSP".

**Reply by Seekuhnetz** [reply]
@kazo0 It looks like you already updated it? It now looks good in a new project i just created.

**Reply by kazo0** [reply]
Interesting! I will still take a look at your uploaded sample to see what may have gone wrong
